templates/service_templates/redis.service.erb

```diff
@@ -1,15 +1,20 @@
 [Unit]
-Description=Advanced key-value store for <%= @title %>
+Description= Redis Advanced key-value store for instance <%= @title %>
 After=network.target
+After=network-online.target
+Wants=network-online.target
 
 [Service]
 Type=forking
-ExecStart=/usr/bin/redis-server <%= @redis_file_name %>
-ExecStop=/usr/bin/redis-server -p <%= @port %> shutdown
+ExecStart=/usr/bin/redis-server <%= @redis_file_name %> --supervised systemd
+ExecStop=/usr/libexec/redis-shutdown <%= @redis_server_name %>
```

Comment:
what about something like `ExecStop=/bin/kill -s TERM $MAINPID` instead? This way redis-shutdown isn't needed and afaics Redis handles SIGTERM fine.